### PR TITLE
[JFG-1067] Separate Master - JaaS/Env and Master - JaaS/Execution communication

### PR DESCRIPTION
**Pull request by a-badaev**

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/JaggerLauncher.java [hunk 1]
```diff
@@ -205,8 +198,48 @@ public void run() {
...
+                                                  environmentProperties.getProperty("realtime.jaas.endpoint")
+            );
+        
+            Optional<TestExecutionEntity> optionalExecution = execApiClient.getExecution();
+            if (!optionalExecution.isPresent()) { // didn't manage to acquire an entity
+                return;
+            }
+            TestExecutionEntity execution = optionalExecution.get();
+            if (execution.getStatus() != TestExecutionEntity.TestExecutionStatus.PENDING) {
+                log.warn("Received execution is not in PENDING state. Going to proceed with next...");
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/JaggerLauncher.java]: Received execution with id {} is not...

File: chassis/core/src/main/java/com/griddynamics/jagger/master/JaasExecApiClient.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    
+    private final URI executionUri;
+    private final RestTemplate restTemplate = new RestTemplate();
+    
+    public JaasExecApiClient(String executionId, String jaasEndpoint) {
+        URI jaasEndpointUri;
+        try {
+            jaasEndpointUri = new URI(jaasEndpoint);
+        } catch (URISyntaxException e) {
+            throw new IllegalStateException("Incorrect JaaS endpoint", e);
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/master/JaasExecApiClient.java]: Incorrect jaas endpoint: {}

File: jaas-client/src/main/java/com/griddynamics/jagger/jaas/storage/model/TestEnvUtils.java [hunk 3]
```diff
@@ -18,6 +18,7 @@ private TestEnvUtils() {
     public static final String EXPIRES_HEADER = PREFIX + "Expires";
     public static final String LOAD_SCENARIO_HEADER = PREFIX + "Load-Scenario-To-Execute";
     public static final String TEST_PROJECT_URL_HEADER = PREFIX + "Test-Project-Url";
```
Comment (dlatnikov) [inline jaas-client/src/main/java/com/griddynamics/jagger/jaas/storage/model/TestEnvUtils.java]: do we need LOAD_SCENARIO_HEADER and TEST_PROJECT_URL_HEADER?

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestExecutionRestController.java [hunk 4]
```diff
@@ -55,8 +57,27 @@ public TestExecutionRestController(TestExecutionService testExecutionService) {
...
     private void validateTestExecution(TestExecutionEntity testExecution) {
+    
+        if (StringUtils.isEmpty(testExecution.getEnvId())) {
+            throw new InvalidTestExecutionException(
+                    "envId must be specified");
+        }
+        
+        if (testExecution.getLoadScenarioId() == null && testExecution.getTestProjectURL() == null) {
+            throw new InvalidTestExecutionException(
+                    "Neither a test project URL nor a load scenario name are not specified");
```
Comment (dlatnikov) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestExecutionRestController.java]: are not specified => are specified

**Comment by a-badaev** [comment]
fixed all comments

**Pull request merged**